crates/uv/src/commands/toolchain/list.rs

```diff
@@ -0,0 +1,68 @@
+use std::fmt::Write;
+use std::ops::Deref;
+
+use anyhow::Result;
+use itertools::Itertools;
+
+use uv_cache::Cache;
+use uv_configuration::PreviewMode;
+use uv_toolchain::downloads::PythonDownloadRequest;
+use uv_toolchain::managed::InstalledToolchains;
+use uv_warnings::warn_user;
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+use crate::settings::ToolchainListIncludes;
+
+/// List available toolchains.
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn list(
```

Comment:
This implementation feels kind of sloppy. cc @snowsignal maybe a good one for you to critique :)